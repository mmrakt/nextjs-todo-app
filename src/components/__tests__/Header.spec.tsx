import MenuIcon from '@material-ui/icons/Menu'
import { cleanup } from '@testing-library/react'
import React from 'react'
import renderer from 'react-test-renderer'

import Drawer from '../header/drawer'
import DrawerItem from '../header/drawerItem'
import MenuList from '../header/menuList'

afterEach(cleanup)

// describe('Header', () => {
//     const component = renderer.create(<Header />)
//     const tree = component.toJSON()

//     expect(tree).toMatchSnapshot()
// })

test('DrawerItem', () => {
  const component = renderer.create(
    <DrawerItem href="/" icon={<MenuIcon />} listItemText="トップメニュー" />
  )
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})

test('Drawer', () => {
  const component = renderer.create(
    <Drawer isOpenDrawer={false} handleClose={jest.fn()} />
  )
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})

test('MenuList', () => {
  const component = renderer.create(<MenuList />)
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})
