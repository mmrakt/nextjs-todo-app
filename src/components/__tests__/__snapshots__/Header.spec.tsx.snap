// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="MuiListItemAvatar-root"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
  <div
    className="MuiListItemText-root"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
    >
      トップメニュー
    </span>
  </div>
</div>
`;

exports[` 2`] = `null`;

exports[`Drawer 1`] = `null`;

exports[`DrawerItem 1`] = `
<div
  aria-disabled={false}
  className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="MuiListItemAvatar-root"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
  <div
    className="MuiListItemText-root"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
    >
      トップメニュー
    </span>
  </div>
</div>
`;
